_source/_posts/2022-06-28-add-auth-to-any-app-with-oauth2-proxy.adoc

```diff
@@ -0,0 +1,583 @@
+<.> Allow all email domains unless you use a social auth provider, you'll want to manage this in your IdP and NOT in your application.
+<.> Set the redirect URL to an `http` URL, this defaults to `https`.
+<.> Open up the `.env` file and set this variable to a random 32-byte base64 string `openssl rand -base64 32 | tr -- '+/' '-_'`
+<.> Expose port `4180`.
+====
+
+Start everything up by running:
+
+[source,bash]
+----
+docker compose up
+----
+
+Now open your browser to `\http://localhost:4180/echo`, and you will be redirected to a page with a "sign-in" button. Click the button,  and you will be redirected back to the "echo" application, and you should see information about the newly authenticated request!
+
+image::{% asset_path 'blog/add-auth-to-any-app-with-oauth2-proxy/sign-in-page.png' %}[alt=Screenshot of oauth2-proxy default sign-in page,width=500,align=center]
+
+TIP: If you are already logged into your Okta account, open an incognito/private browser to see the full sign-in flow.
+
+Great, the application is now secured, but we still have a few things to clean up:
+
+* All the session state is stored in a cookie.
+* The initial double redirect sign-in page has to go.
+* We haven't talked about API access yet.
+
+These first two problems can be fixed with a few updates to the OAuth2 Proxy configuration. Edit the `docker-compose.yml` file:
+
+====
+[source,diff]
+----
+       OAUTH2_PROXY_COOKIE_SECRET: ${OAUTH2_PROXY_COOKIE_SECRET}
++      OAUTH2_PROXY_SKIP_PROVIDER_BUTTON: true # <.>
++      OAUTH2_PROXY_COOKIE_NAME: SESSION # <.>
++      OAUTH2_PROXY_COOKIE_SAMESITE: lax # <.>
++      OAUTH2_PROXY_SESSION_STORE_TYPE: redis # <.>
++      OAUTH2_PROXY_REDIS_CONNECTION_URL: redis://redis
+    ports:
+      - 4180:4180
++    depends_on:
++      - redis
++
++  redis:# <.>
++    image: redis:7.0.2-alpine3.16
++    volumes:
++      - cache:/data # <.>
++
++volumes:
++  cache:
++    driver: local
+----
+
+<.> Skip the default login page and redirect directly to the IdP.
+<.> By default, the cookie name is `_oauth2_proxy`; change it to `SESSION`
+<.> Set the cookie's same site policy to `lax`; the redirect from the OAuth IdP will need the session cookie.
+<.> Use Redis to store session information
+<.> Start up a Redis container
+<.> Persist the Redis data between restarts
+====
+
+Stop the docker-compose process (kbd:[Ctrl + C]) and start it up again:
+
+[source,bash]
+----
+docker compose up
+----
+
+Once again, open your browser to `\http://localhost:4180/` and open up your network tab, you will see the renamed and now smaller `SESSION` cookie.
+
+You _could_ stop here, but you shouldn't; we still have a few issues: API clients are not supported, and we haven't talked about logging out.
+
+IMPORTANT: For the next section, you will need an access token. You can use the access token from your last request's `x-access-token` header. Open your terminal and set an environment variable: `export TOKEN=\{you-token-value}`.
```

Comment:
is the `\` suppose to be there?

Reply:
Yeah, IIRC there is some odd interaction with liquid tags and adoc.

Reply:
Also, should "you-token-value" be "your-token-value" for readability?